backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -536,24 +548,38 @@ public RepositoryEntity getRepositoryById(Session session, RepositoryIdentificat
   }
 
   @Override
-  public ListBranchCommitsRequest.Response listBranchCommits(ListBranchCommitsRequest request)
+  public ListCommitsLogRequest.Response listCommitsLog(ListCommitsLogRequest request)
       throws ModelDBException {
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
       session.beginTransaction();
       RepositoryEntity repository = getRepositoryById(session, request.getRepositoryId());
 
-      Query query = session.createQuery(CHECK_BRANCH_IN_REPOSITORY_HQL);
-      query.setParameter("repositoryId", repository.getId());
-      query.setParameter("branch", request.getBranch());
-      BranchEntity branchEntity = (BranchEntity) query.uniqueResult();
-      if (branchEntity == null) {
-        throw new ModelDBException(ModelDBConstants.BRANCH_NOT_FOUND, Code.NOT_FOUND);
+      String referenceCommit;
+
+      if (request.getBranch() != "") {
+        Query query = session.createQuery(CHECK_BRANCH_IN_REPOSITORY_HQL);
+        query.setParameter("repositoryId", repository.getId());
+        query.setParameter("branch", request.getBranch());
+        BranchEntity branchEntity = (BranchEntity) query.uniqueResult();
+        if (branchEntity == null) {
+          throw new ModelDBException(ModelDBConstants.BRANCH_NOT_FOUND, Code.NOT_FOUND);
+        }
+        referenceCommit = branchEntity.getCommit_hash();
+      } else {
+        Query query = session.createQuery(CHECK_COMMIT_IN_REPOSITORY_HQL);
+        query.setParameter("repositoryId", repository.getId());
+        query.setParameter("commit", request.getCommitSha());
+        CommitEntity commit = (CommitEntity) query.uniqueResult();
+        if (commit == null) {
+          throw new ModelDBException(ModelDBConstants.COMMIT_NOT_FOUND, Code.NOT_FOUND);
+        }
+        referenceCommit = commit.getCommit_hash();
       }
```

Comment:
since you have the PK : CommitEntity commit = session.get(CommitEntity.class, request.getCommitShaA());

Reply:
Done.